specification/marketplacecatalog/resource-manager/Products/preview/2022-09-01-preview/products.json

```diff
@@ -0,0 +1,1411 @@
+          "ListingSubscriptionBasedPrivateProducts"
+        ],
+        "summary": "Returns a list of public, private and preview products, by subscription as audience",
+        "operationId": "ListingSubscriptionBasedPrivateProducts_Get",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "type": "string",
+            "name": "api-version",
+            "x-originalName": "apiVersion",
+            "in": "query",
+            "required": true,
+            "description": "Version of the REST API used for the request. ",
+            "default": "2022-09-01-preview",
+            "x-nullable": true
+          },
+          {
+            "type": "string",
+            "name": "language",
+            "in": "query",
+            "description": "Language to search, ISO 639-1 two-letter code, possible values - 'en,cs,de,es,fr,hu,it,ja,ko,nl,pl,pt-br,pt-pt,ru,sv,tr,zh-hans,zh-hant'. Default='en'",
+            "default": "en",
+            "x-nullable": true
+          },
+          {
+            "type": "string",
+            "name": "storefront",
+            "in": "query",
+            "description": "Relevant storefront, possible values - 'azure, amp, appsource'. Default='azure'",
+            "default": "azure",
+            "x-nullable": true
+          },
+          {
+            "type": "string",
+            "name": "market",
+            "in": "query",
+            "description": "Product sold market, possible values at https://docs.microsoft.com/en-us/azure/marketplace/marketplace-geo-availability-currencies, such as ['US']",
+            "x-nullable": true
+          },
+          {
+            "type": "boolean",
+            "name": "excludePublic",
+            "x-originalName": "excludePublicOffersAndPublicPlans",
+            "in": "query",
+            "description": "Denotes whether to exclude public products and plans",
+            "default": false,
+            "x-nullable": false
+          },
+          {
+            "type": "string",
+            "name": "$select",
+            "in": "query",
+            "description": "Selects which properties to include in the results. Example: '$select=displayName'"
+          },
+          {
+            "type": "string",
+            "name": "$filter",
+            "in": "query",
+            "description": "Filters the results, based on a Boolean condition. Example: '$filter=(isReseller eq true)'"
+          },
+          {
+            "type": "string",
+            "name": "$expand",
+            "in": "query",
+            "description": "Expands related entities inline. Example: '$expand=startingPrice'"
+          },
+          {
+            "type": "string",
+            "name": "$orderBy",
+            "in": "query",
+            "description": "Ordering expression for the results using OData notation. Example: '$orderby=displayName desc'"
+          }
+        ],
+        "responses": {
```

Comment:
>responses [](http://example.com/codeflow?start=9&length=9)

Can there be a large number of products returned, which would need paginated response? Adding pagination to GA'ed resource, is breaking change.

Use [https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-pageable](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-pageable) for pagination.

Reply:
"CatalogApiResponse" is our own custom implementation of paged response (not that along with the array of items we also return a next page link where relevant)